testproject/jijtest/build.gradle

```diff
@@ -2,14 +2,30 @@ plugins {
     id 'net.neoforged.moddev'
 }
 
+sourceSets {
+    plugin
+}
+
+def pluginJar = tasks.register("pluginJar", Jar) {
+    from(sourceSets.plugin.output)
+    archiveClassifier = "plugin"
+    manifest {
+        attributes(
+            'FMLModType': 'LIBRARY',
+            'Automatic-Module-Name': 'jij.plugin'
+        )
+    }
+}
+
 evaluationDependsOn(":coremod") // Because of the sourceset reference
 
 dependencies {
     implementation project(":coremod")
-
+    implementation files(sourceSets.plugin.output)
```

Comment:
Why implementation?

Reply:
Hmm it's probably better to add it to the compile and runtime CPs.